recipes/publish.rb

```diff
@@ -29,44 +29,19 @@
 
   # set local variables we're going to use in `lazy` properties later in
   # the chef run
-  artifact = nil
   build_version = nil
-  last_build_env = nil
   project_name = node['delivery']['change']['project']
 
-  execute 'build-plan' do
-    command "unset TERM; HAB_ORIGIN=#{origin} sudo -E #{hab_binary} studio" \
-            " -r #{hab_studio_path}" \
-            " build #{habitat_plan_dir}"
-    cwd node['delivery']['workspace']['repo']
-    live_stream true
-  end
-
-  ruby_block 'load-build-output' do
-    block do
-      last_build_env = Hash[*::File.read(::File.join(hab_studio_path,
-                                                     'src/results/last_build.env')).split(/[=\n]/)]
-
-      artifact = last_build_env['pkg_artifact']
-      build_version = [last_build_env['pkg_version'], last_build_env['pkg_release']].join('-')
-    end
-  end
-
+  # Only build and publish if we have a depot token
   if habitat_depot_token?
-    depot_token = project_secrets['habitat']['depot_token']
-
-    execute 'upload-pkg' do
-      command lazy {
-        "#{hab_binary} pkg upload" \
-        " --url #{node['habitat-build']['depot-url']}" \
-        " #{hab_studio_path}/src/results/#{artifact}"
-      }
-      env(
-        'HOME' => delivery_workspace,
-        'HAB_AUTH_TOKEN' => depot_token
-      )
-      live_stream true
-      sensitive true
+    habitat project_name do
+      origin origin
+      plan_dir habitat_plan_dir
+      cwd node['delivery']['workspace']['repo']
+      home_dir delivery_workspace
+      auth_token project_secrets['habitat']['depot_token']
+      url node['habitat-build']['depot-url']
+      action :build_and_publish
```

Comment:
With the other changes I'm suggesting, your action would look like this:

```ruby
action [:build, :publish]
```

This is idiomatic and expected behavior for Chef users. For example, commonly a `service` resource will be written like:

```ruby
service "food" do
  action [:enable, :start]
end
```